float64 w0
float64 w1
float64 dw0
float64 dw1
float64 ddw0
float64 ddw1
